src/test/java/org/gaul/s3proxy/ShardedBlobStoreTest.java

```diff
@@ -0,0 +1,268 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gaul.s3proxy;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.InputStream;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.ByteSource;
+import com.google.inject.Module;
+
+import org.jclouds.ContextBuilder;
+import org.jclouds.apis.Storage;
+import org.jclouds.blobstore.BlobStore;
+import org.jclouds.blobstore.BlobStoreContext;
+import org.jclouds.blobstore.domain.Blob;
+import org.jclouds.blobstore.domain.PageSet;
+import org.jclouds.blobstore.domain.StorageMetadata;
+import org.jclouds.blobstore.options.CopyOptions;
+import org.jclouds.logging.slf4j.config.SLF4JLoggingModule;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public final class ShardedBlobStoreTest {
+    private int shards;
+    private String prefix;
+    private String containerName;
+    private BlobStoreContext context;
+    private BlobStore blobStore;
+    private BlobStore shardedBlobStore;
+
+    @Before
+    public void setUp() {
+        containerName = TestUtils.createRandomContainerName();
+        shards = 10;
+        prefix = TestUtils.createRandomContainerName();
+        context = ContextBuilder
+                .newBuilder("transient")
+                .credentials("identity", "credential")
+                .modules(ImmutableList.<Module>of(new SLF4JLoggingModule()))
+                .build(BlobStoreContext.class);
+        blobStore = context.getBlobStore();
+        ImmutableMap<String, Integer> shardsMap =
+                new ImmutableMap.Builder<String, Integer>()
+                        .put(containerName, shards).build();
+        ImmutableMap<String, String> prefixesMap =
+                new ImmutableMap.Builder<String, String>()
+                        .put(containerName, prefix).build();
+        shardedBlobStore = ShardedBlobStore.newShardedBlobStore(
+                blobStore, shardsMap, prefixesMap);
+    }
+
+    @After
+    public void tearDown() {
+        if (context != null) {
+            blobStore.list().stream()
+                    .filter(sm -> sm.getName().startsWith(prefix))
+                    .forEach(sm -> blobStore.deleteContainer(sm.getName()));
+            context.close();
+        }
+    }
+
+    @Test
+    public void testCreateContainer() {
+        assertThat(shardedBlobStore.createContainerInLocation(
+                null, containerName)).isTrue();
+        assertThat(blobStore.containerExists(containerName)).isFalse();
+        PageSet<? extends StorageMetadata> listing = blobStore.list();
+        List<? extends StorageMetadata> shardsBuckets = listing.stream()
+                .filter(sm -> sm.getName().startsWith(prefix)).collect(
+                        Collectors.toList());
```

Comment:
These are more commonly written with each pipeline stage on their own lines for readability:

```java
        List<? extends StorageMetadata> shardsBuckets = listing.stream()
                .filter(sm -> sm.getName().startsWith(prefix))
                .collect(Collectors.toList());
```

Reply:
Replaced with a loop.